app/services/product_tower_service.py

```
@@ -0,0 +1,272 @@
+    
+    def _normalize_beauty_terms(self, text: str) -> str:
+        """뷰티 관련 용어 정규화 (중복 방지)"""
+        # 설정에서 뷰티 용어 매핑 가져오기
+        beauty_terms = settings.BEAUTY_TERMS_MAPPING
+        
+        # 길이 순으로 정렬하여 긴 키워드부터 처리 (중복 방지)
+        sorted_terms = sorted(beauty_terms.items(), key=lambda x: len(x[0]), reverse=True)
+        
+        for original, normalized in sorted_terms:
+            # 단어 경계를 고려한 정확한 매칭
+            import re
```

Flake8 (7.2.0)

[error] import 're' from line 1 shadowed by loop variable

(F402)